### PR TITLE
Refactor | Move typing to single directory

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
I recommend using Github's codespace for a quick code review. (Hit period on this screen).

Basically all this PR does is move all typing throughout the app to a single `types` directory.  to keep things less confusing this PR only moves the typing. no other changes were made.

This move is first step towards a better data-layer. Any 'exported' type declaration should be placed in this directory to help prevent duplicating types, as well as cleaning up the typing imports on each component.

### CODE REVIEW

File: src/types/daoProposal.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export enum UsulVoteChoice {
+  No,
+  Yes,
+  Abstain,
+}
+
+export enum ProposalIsPassedError {
+  MAJORITY_YES_VOTES_NOT_REACHED = 'majority yesVotes not reached',
+  QUORUM_NOT_REACHED = 'a quorum has not been reached for the proposal',
+  PROPOSAL_STILL_ACTIVE = 'voting period has not passed yet',
```
Comment (herbig) [inline src/types/daoProposal.ts:77]: @tbwebb22 we match these errors with the error messages in the contracts, so we should make a note to update error messages in the frontend as well

**Review by herbig — APPROVED** [review]

File: src/types/fractal.ts [hunk 2]
```diff
@@ -1,58 +1,75 @@
...
+    treasuryDispatch: React.Dispatch<TreasuryActions>;
+    governanceDispatch: React.Dispatch<GovernanceActions>;
+    gnosisDispatch: React.Dispatch<GnosisActions>;
+  };
+  actions: {
+    refreshSafeData: () => Promise<void>;
+    lookupModules: (_moduleAddresses: string[]) => Promise<IGnosisModuleData[] | undefined>;
+    getVetoGuardContracts: (
+      _guardAddress: string,
+      _modules?: IGnosisModuleData[] | undefined
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/types/fractal.ts:53]: ha! good catch. I'll note for the future refactor.

File: src/types/fractal.ts [hunk 3]
```diff
@@ -1,58 +1,75 @@
...
-  moduleContract: FractalUsul | FractalModule | undefined;
-  moduleAddress: string;
-  moduleType: GnosisModuleType;
+export interface IFractalContext {
+  gnosis: IGnosis;
+  treasury: ITreasury;
+  governance: IGovernance;
+  account: IConnectedAccount;
+  dispatches: {
+    treasuryDispatch: React.Dispatch<TreasuryActions>;
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/types/fractal.ts:44]: yeah I agree. I'll not for the next coming PRs

File: src/types/createProposal.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  proposalData: ProposalExecuteData | undefined;
+  nonce: number | undefined;
+  successCallback: () => void;
+  isCreateDisabled: boolean;
+  transactions: TransactionData[];
+  setTransactions: Dispatch<SetStateAction<TransactionData[]>>;
+  removeTransaction: (transactionNumber: number) => void;
+}
+
+export interface UsulMetadataProps {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/types/createProposal.ts:35]: I believe so. I'll check fo rthis

File: src/providers/Fractal/hooks/useGnosisApiServices.ts [hunk 5]
```diff
@@ -1,29 +1,22 @@
...
-import { GnosisAction, TreasuryAction } from '../constants/actions';
-import { GnosisActions, IGnosis, TreasuryActions } from '../types';
+import {
+  IGnosis,
+  TreasuryActions,
+  GnosisActions,
+  GnosisAction,
+  TreasuryAction,
+  AllTransfersListResponse,
+} from '../../../types';
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/providers/Fractal/hooks/useGnosisApiServices.ts:15]: god call 👍 that could help clean things up

File: src/components/ProposalCreate/UsulMetadata.tsx [hunk 6]
```diff
@@ -1,25 +1,9 @@
 import { Button, Divider, VStack } from '@chakra-ui/react';
-import { Dispatch, SetStateAction, useState } from 'react';
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { UsulMetadataProps } from '../../types';
 import { InputComponent, TextareaComponent } from './InputComponent';
 
-export interface UsulMetadataProps {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/components/ProposalCreate/UsulMetadata.tsx]: opps, you are right, I moved this one on accident. you are correct single source of truth for multi-use types

**Pull request merged**